site/docs/how-to/envs.md

```diff
@@ -0,0 +1,37 @@
+---
+sidebar_position: 4
+slug: '/envs'
+---
+
+# Replicate changes between dev, staging, and prod environments
+
+Due to capacity, region, and other differences across cloud infrastructure environments, we recommend that IaSQL users have one AWS account per environment (dev, staging, prod, etc). Each AWS account is [connected](./aws.mdx) to an [IaSQL database](../concepts/db.md) that is modified independently.
+
+Complex, or delicate, infrastructure changes should be treated as transactions. [IaSQL transactions](../concepts/transaction.md) are a bit different from normal DB transactions in that they are multiuser and can be previewed, with the function [`iasql_preview`](../modules/builtin/iasql_functions.md), for the changes that the ongoing transaction will create in the cloud similar to Pulumi’s preview or Terraform’s plan.
+
+```sql
+--- starts a transaction
+SELECT * FROM iasql_begin();
+
+--- infrastructure changes
+
+--- calls iasql_preview to see the result in the cloud
+select * from iasql_preview();
+
+--- more infrastructure changes
+
+--- calls iasql_preview to see the result in the cloud
+select * from iasql_preview();
+
+--- calls iasql_commit should create
+select * from iasql_commit();
+```
+
+Once the transaction is committed successfully, IaSQL automatically rolls back failed transactions, and the desired changes are committed into the DB for one environment, we have a PG function that looks at the audit log and generates the SQL representing changes done from a given point in time.
```

Comment:
Here the first sentence says "Once the transaction is committed successfully, IaSQL automatically rolls back failed **transactions**", which seems like the second transaction is something else than the first. I think this sentence can be written more clearly to show the unsuccessful transactions are automatically rolled back.

Reply:
Do we have partial apply? Or a transaction is either 1 (applied) or 0 (rolled back)?

Reply:
good point, fixed